src/UrlDataFetcher.js

```diff
@@ -0,0 +1,18 @@
+import React, {useEffect} from "react";
+import $ from "jquery";
+
+export default function UrlDataFetcher({url, setData}) {
+    useEffect(() => {
+        setData(null);
+        const xhr = $.getJSON(url)
+            .done((data) => setData(data))
+            .fail(setData(null));
+
+        return function abort() {
+            xhr.abort();
+            setData(null);
+        }
+    }, [url]);
+
+    return (<></>);
```

Comment:
What you are doing here resembles a lot to custom hooks functionality.
However, it is not a good solution to create a component like this that returns nothing.
Since there is no presentation it should be either a HOC or the functionality should be inside the main component.